src/logger.ts

```diff
@@ -7,12 +7,19 @@ export function setVerbose(value: boolean)
     isVerbose = value;
 }
 
-export function warn(msg: string, data?: any)
+export function warn(msg: string, data?: any|TAnyDoclet)
 {
+
     if (typeof(console) === 'undefined')
         return;
 
-    console.warn(`${header} ${msg}`);
+    let prefix = header;
+    if (arguments.length > 1 && data && data.hasOwnProperty('meta')) {
```

Comment:
```suggestion
    if (data && data.meta)
    {
```

Don't need to check `arguments.length` and `hasOwnProperty` is overkill here.